docs/configuring-a-resource.md

```diff
@@ -22,126 +22,239 @@ cloud provider; sometimes it could simply be the name of resource
+
+Here are [the types for the External Name configuration]:
 
 ```go
+// SetIdentifierArgumentsFn sets the name of the resource in Terraform attributes map,
+// i.e. Main HCL file.
+type SetIdentifierArgumentsFn func(base map[string]interface{}, externalName string)
+// GetExternalNameFn returns the external name extracted from the TF State.
+type GetExternalNameFn func(tfstate map[string]interface{}) (string, error)
+// GetIDFn returns the ID to be used in TF State file, i.e. "id" field in
+// terraform.tfstate.
+type GetIDFn func(ctx context.Context, externalName string, parameters map[string]interface{}, providerConfig map[string]interface{}) (string, error)
+
 // ExternalName contains all information that is necessary for naming operations,
 // such as removal of those fields from spec schema and calling Configure function
 // to fill attributes with information given in external name.
 type ExternalName struct {
-    // SetIdentifierArgumentFn sets the name of the resource in Terraform argument
-    // map.
-    SetIdentifierArgumentFn SetIdentifierArgumentFn
-    
-    // OmittedFields are the ones you'd like to be removed from the schema since
-    // they are specified via external name. You can omit only the top level fields.
-    // No field is omitted by default.
-    OmittedFields []string
-    
-    // DisableNameInitializer allows you to specify whether the name initializer
-    // that sets external name to metadata.name if none specified should be disabled.
-    // It needs to be disabled for resources whose external name includes information
-    // more than the actual name of the resource, like subscription ID or region
-    // etc. which is unlikely to be included in metadata.name
-    DisableNameInitializer bool
+  // SetIdentifierArgumentFn sets the name of the resource in Terraform argument
+  // map. In many cases, there is a field called "name" in the HCL schema, however,
+  // there are cases like RDS DB Cluster where the name field in HCL is called
+  // "cluster_identifier". This function is the place that you can take external
+  // name and assign it to that specific key for that resource type.
+  SetIdentifierArgumentFn SetIdentifierArgumentsFn
+  
+  // GetExternalNameFn returns the external name extracted from TF State. In most cases,
+  // "id" field contains all the information you need. You'll need to extract
+  // the format that is decided for external name annotation to use.
+  // For example the following is an Azure resource ID:
+  // /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/mygroup1
+  // The function should return "mygroup1" so that it can be used to set external
+  // name if it was not set already.
+  GetExternalNameFn GetExternalNameFn
+  
+  // GetIDFn returns the string that will be used as "id" key in TF state. In
+  // many cases, external name format is the same as "id" but when it is not
+  // we may need information from other places to construct it. For example,
```

Comment:
This adds to the equivocation that I mentioned in https://github.com/crossplane/terrajet/pull/187/files#r777662083

Perhaps a section of the docs needs to be referenced that can go into more detail on the distinctions.

Reply:
I missed the example cases describing these scenarios later in the doc.  

It may be clearer to discuss the external name vs name vs IDs topic first before introducing these go fields.  

After that, the comments in these go fields may still be confusing. Perhaps the type can be offered in these docs without the comments (show the simple struct and link to the source for details). The text and examples in this documentation can describe how the fields can be used, and expand on usage within the scenarios and examples offered.